### PR TITLE
Change init functions to use assignment statements

### PR DESCRIPTION
**Pull request by lankavitharana**
This PR will validate the constant assignments as well

### CODE REVIEW

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java [hunk 1]
```diff
@@ -466,6 +469,12 @@ public void visit(BLangAssignment assignNode) {
             }
             ((BLangVariableReference) varRef).lhsVar = true;
             expTypes.add(typeChecker.checkExpr(varRef, env).get(0));
+            if (varRef instanceof  BLangSimpleVarRef && ((BLangVariableReference) varRef).symbol.flags == Flags.CONST
+                    && env.enclInvokable != env.enclPkg.initFunction) {
+                dlog.error(varRef.pos, DiagnosticCode.INVALID_VARIABLE_ASSIGNMENT, varRef);
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java]: Let's use a meaningful error here. We've used the following error in the previous compiler.

cannot.assign.value.constant = cannot assign a value to constant ''{0}''

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 2]
```diff
@@ -516,8 +516,7 @@ private void defineServiceMembers(List<BLangService> services, SymbolEnv pkgEnv)
         services.forEach(service -> {
             SymbolEnv serviceEnv = SymbolEnv.createServiceEnv(service, service.symbol.scope, pkgEnv);
             service.vars.forEach(varDef -> defineNode(varDef.var, serviceEnv));
-            defineServiceInitFunction(service);
-            defineNode(service.initFunction, serviceEnv);
+            defineServiceInitFunction(service, serviceEnv);
 //            service.symbol.initFunctionSymbol = service.initFunction.symbol;
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Shall we get rid of comments? :)

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileWriter.java [hunk 3]
```diff
@@ -303,15 +303,6 @@ private static void writeConnectorInfo(DataOutputStream dataOutStream,
         // TODO write property flags  e.g. public
         dataOutStream.writeInt(connectorInfo.signatureCPIndex);
 
-//        Map<Integer, Integer> methodTable = connectorInfo.getMethodTableIndex();
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileWriter.java]: Let's get rid of comments :)

Comment (lankavitharana) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileWriter.java]: it's already removed in my local and synced with this branch, not sure how it appears here :(

**Pull request merged**